sdk/keyvault/keyvault-certificates/src/lro/operation/operation.ts

```diff
@@ -4,17 +4,32 @@
 import { AbortSignalLike } from "@azure/abort-controller";
 import { PollOperationState, PollOperation } from "@azure/core-lro";
 import { RequestOptionsBase } from "@azure/core-http";
-import { CertificateClientInterface, CertificateOperation } from "../../certificatesModels";
+import {
+  CertificateOperation,
+  CertificateClientInterface,
+  KeyVaultCertificateWithPolicy
+} from "../../certificatesModels";
 
 /**
- * An interface representing the state of a create certificate's poll operation
+ * An interface representing the publicly available properties of the state of the CertificateOperationPoller.
  */
-export interface CertificateOperationPollOperationState
-  extends PollOperationState<CertificateOperation> {
+export interface CertificateOperationPollOperationPublicState
+  extends PollOperationState<KeyVaultCertificateWithPolicy> {
   /**
    * The name of the certificate.
    */
   certificateName: string;
+  /**
+   * The operation of the certificate
+   */
+  certificateOperation?: CertificateOperation;
+}
+
+/**
+ * An interface representing the state of a create certificate's poll operation
+ */
+export interface CertificateOperationPollOperationPrivateState
+  extends CertificateOperationPollOperationPublicState {
```

Comment:
Why does the private state extend the public state?  I would imagine these are separate and private state is mostly hidden from the user?

Reply:
Private state is indeed hidden from the user. The user only gets the public state.